doc/python/px-arguments.md

```diff
@@ -101,33 +115,55 @@ fig.show()
 Many more examples of wide-form and messy data input can be found in our [detailed wide-form support documentation](/python/wide-form/).
 
 
-### Input Data as Pandas `DataFrame`s
+## Dataframe Input
+
+The first argument of every `px` function is `data_frame`. If you provide a dataframe as a `px` function's first argument, you can then specify column names as strings from the dataframe as other arguments.
+
+### Supported Dataframes
+
+`px` functions natively support pandas, Polars, and PyArrow dataframes. `px` uses [Narwhals](https://narwhals-dev.github.io/narwhals/) to provide this native dataframe support. Other types of dataframes that are currently supported by Narwhals, for example cuDF and Modin, may also work with `px`.
+
+You can also pass dataframes that are not natively supported, but which support the [dataframe interchange protocol](https://data-apis.org/dataframe-protocol/latest/).
 
-As shown above, `px` functions supports natively pandas DataFrame. Arguments can either be passed as dataframe columns, or as column names if the `data_frame` argument is provided.
+PySpark dataframes are also supported and are converted to pandas dataframes internally by Plotly Express.
 
-#### Passing columns as arguments
+> There is a known issue where `px.icicle`, `px.treemap`, and `px.sunburst` used with a `path` parameter won't work with cuDF dataframes.
+
+#### Additional Dependencies Required
+
+- To use Polars, PyArrow, or other dataframes supported natively with Plotly Express, you'll need to have NumPy installed. You can install it with: `pip install numpy`.
```

Comment:
A little misleading, since NumPy is required for Pandas dataframes as well, it's just that NumPy is a dependency of Pandas so it's likely to be already installed if you're instantiating a Pandas dataframe.

Reply:
That's a fair point. Updated